usecases/modulecomponents/batch/batch.go

```diff
@@ -65,18 +66,22 @@ func (b BatchJob) copy() BatchJob {
 	}
 }
 
-type BatchClient interface {
+type BatchClient[T []float32 | [][]float32] interface {
```

Comment:
you are sometimes using types.Vector as type restriction, can you also use this here?